File: OCH_Tao/ABC/B/185.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, , T] = input.shift().split(" ").map(x => parseInt(x, 10));
let cnt = N;
let X = 0;
for (const i of input) {
  const [A, B] = i.split(" ").map(x => parseInt(x, 10));
  cnt = Math.max(0, cnt - (A - X));
  if (cnt > 0) {
    cnt = Math.min(N, cnt + (B - A));
    X = B;
  } else {
    console.log("No");
    return false;
  }
}
cnt = Math.max(0, cnt - (T - X));
console.log(cnt > 0 ? "Yes" : "No");

File: OCH_Tao/ABC/B/185.py
N,M,T = map(int,input().split())
cnt = N
X = 0
for i in range(M):
  A,B = map(int,input().split())
  cnt=max(0,cnt-(A-X))
  if cnt>0:
    cnt=min(N,cnt+(B-A))
    X=B
  else:
    print("No")
    break
else:
  cnt=max(0,cnt-(T-X))
  if cnt>0:
    print("Yes")
  else:
    print("No")